### PR TITLE
Moving the LLVM branch up.

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: vendor/llvm/llvm.bzl [hunk 1]
```diff
@@ -214,7 +215,7 @@ posix_cmake_vars = {
     "HAVE_GETTIMEOFDAY": 1,
     "HAVE_INT64_T": 1,
     "HAVE_ISATTY": 1,
-    "HAVE_LIBEDIT": 0,
+    "HAVE_LIBEDIT": 1,
```
Comment (flaub) [inline vendor/llvm/llvm.bzl:218]: We need to have `HAVE_LIBEDIT` needs to be `0`.

File: vendor/llvm/llvm.bzl [hunk 2]
```diff
@@ -312,13 +312,36 @@ llvm_all_cmake_vars = select({
...
         ),
     ),
+    "@bazel_tools//src/conditions:linux_ppc64le": cmake_var_string(
+        _dict_add(
+            cmake_vars,
+            llvm_target_cmake_vars("PowerPC", "powerpc64le-unknown-linux_gnu"),
+            posix_cmake_vars,
+            linux_cmake_vars,
+        ),
+    ),
```
Comment (flaub) [inline vendor/llvm/llvm.bzl:322]: Drop all of these other platforms

File: vendor/mlir/mlir.BUILD [hunk 3]
```diff
@@ -3042,7 +3077,6 @@ cc_library(
         "include/mlir/ExecutionEngine/CRunnerUtils.h",
     ],
     includes = ["include"],
-    local_defines = ["mlir_c_runner_utils_EXPORTS"],
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3045]: We need this for windows, revert this

File: vendor/mlir/mlir.BUILD [hunk 4]
```diff
@@ -2975,11 +3012,9 @@ cc_library(
 
 cc_library(
     name = "AllPassesAndDialects",
-    srcs = ["@org_tensorflow//third_party/mlir:mlir-auto-init.cpp"],
     deps = [
         ":AllPassesAndDialectsNoRegistration",
     ],
-    alwayslink = 1,
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:2982]: This needs to be reverted

File: vendor/mlir/mlir.BUILD [hunk 5]
```diff
@@ -3054,11 +3088,9 @@ cc_library(
         "include/mlir/ExecutionEngine/RunnerUtils.h",
     ],
     includes = ["include"],
-    local_defines = ["mlir_runner_utils_EXPORTS"],
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3057]: revert this

File: vendor/mlir/mlir.BUILD [hunk 6]
```diff
@@ -3054,11 +3088,9 @@ cc_library(
...
     deps = [
         ":mlir_c_runner_utils",
     ],
-    alwayslink = 1,
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3061]: revert this

File: vendor/mlir/mlir.BUILD [hunk 7]
```diff
@@ -3073,99 +3105,99 @@ cc_binary(
...
+        ":GPUToVulkanTransforms",
+        ":GPUTransforms",
+        ":MlirJitRunner",
+        ":Pass",
+        ":SPIRVDialect",
+        ":StandardToLLVM",
+        ":StandardToSPIRVTransforms",
+        "@llvm-project//llvm:Support",
+    ],
+)
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3200]: revert this

File: vendor/llvm/llvm.bzl [hunk 8]
```diff
@@ -312,13 +312,36 @@ llvm_all_cmake_vars = select({
...
+    "@bazel_tools//src/conditions:linux_s390x": cmake_var_string(
+        _dict_add(
+            cmake_vars,
+            llvm_target_cmake_vars("SystemZ", "systemz-unknown-linux_gnu"),
+            posix_cmake_vars,
+            linux_cmake_vars,
+        ),
+    ),
```
Comment (flaub) [inline vendor/llvm/llvm.bzl:344]: drop this

File: vendor/llvm/llvm.bzl [hunk 9]
```diff
@@ -331,6 +354,7 @@ llvm_all_cmake_vars = select({
 
 llvm_linkopts = select({
     "@bazel_tools//src/conditions:windows": [],
+    "@bazel_tools//src/conditions:freebsd": ["-ldl", "-lm", "-lpthread", "-lexecinfo"],
```
Comment (flaub) [inline vendor/llvm/llvm.bzl:357]: drop this

File: vendor/llvm/llvm.bzl [hunk 10]
```diff
@@ -312,13 +312,36 @@ llvm_all_cmake_vars = select({
             darwin_cmake_vars,
         ),
     ),
+    # "@bazel_tools//src/conditions:linux_ppc64le": cmake_var_string( ### PlaidML-change removed
```
Comment (flaub) [inline vendor/llvm/llvm.bzl:315]: I think we should just drop this but leave a comment

File: vendor/mlir/mlir.BUILD [hunk 11]
```diff
@@ -3042,7 +3078,7 @@ cc_library(
         "include/mlir/ExecutionEngine/CRunnerUtils.h",
     ],
     includes = ["include"],
-    local_defines = ["mlir_c_runner_utils_EXPORTS"],
+    local_defines = ["mlir_c_runner_utils_EXPORTS"],  ### PlaidML-change - Windows build
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3081]: nit: can you change the comment to simply: `# PlaidML`

File: vendor/mlir/mlir.BUILD [hunk 12]
```diff
@@ -3073,7 +3107,7 @@ cc_binary(
     ],
 )
 
-# cc_binary(
+# cc_binary( ###PlaidML-change Removed
```
Comment (flaub) [inline vendor/mlir/mlir.BUILD:3110]: I think we should mark this for each rule.

**Review by LLITCHEV — APPROVED** [review]
Looks fine to me.

**Pull request merged**